internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/PartitionStatus.java

```diff
@@ -89,7 +89,7 @@ public List<StatusSnapshot> getReplicaHistoricStatusList(String instanceId) {
 
   public boolean hasFatalDataValidationError() {
     for (ReplicaStatus replicaStatus: replicaStatusMap.values()) {
-      if (ExecutionStatus.isError(replicaStatus.getCurrentStatus())
+      if (ExecutionStatus.isError(replicaStatus.getCurrentStatus()) && replicaStatus.getIncrementalPushVersion() != null
```

Comment:
Thanks for the catch. Is it unit testable?

Reply:
Yeah. It's easy. Added

Reply:
nit: can you use StringUtils::isEmpty instead of null check?